include/boost/math/differentiation/autodiff.hpp

```diff
@@ -659,16 +671,17 @@ template<typename RealType, size_t Order>
 template<typename RealType2, size_t Order2>
 fvar<RealType,Order>& fvar<RealType,Order>::operator*=(const fvar<RealType2,Order2>& cr)
 {
+  using ssize_t = std::make_signed<std::size_t>::type;
```

Comment:
Since `std::array<>::difference_type` is signed, this was triggering warnings in MSVC and Clang (with all the all the warnings enabled) due to implicit signed-ness conversions.

Reply:
It seems `ssize_t` has another definition in which the minimum value may be `-1`. I wonder if using the full `std::array<>::difference_type` type is better. See https://stackoverflow.com/a/15739705 I welcome your thoughts - I don't have a conclusion either way.

Reply:
I think that's probably a better idea than assuming the difference type is a signed `std::size_t`, since we're specifically adding to `std::array`'s iterators.

Reply:
The alias `ssize_t` has been changed to `diff_t`; since even though `ssize_t` isn't part of the standard (as far as I've ever read it), many major compilers provide a macro or a type with this name.

I defined `diff_t` as `typename std::array<RealType, Order+1>::difference_type;`. I considered using `decltype(v)` in place of `std::array<RealType, Order+1>`, but I think using `std::array` is more descriptive of the type being aliased than is `decltype(v)`. I used `Order+1` to be consistent with the actual type of the `std::array` within a specialized `fvar` type.

The code has been committed.